docs/session_gathering/session_gathering.md

```diff
@@ -0,0 +1,213 @@
+
+The anatomy of the recommendation is following:
+Recommendation:
+- user
+- unique_id
+- timestamp
+- visit_id
+- page_id
+- panel_id
+- engine_version
+- services
+- search_data
+
+where most of the fields are similar to the user action and have the same meaning but there are also new ones:
+- engine_version: each recommendation can be handled with the chosen recommendation algorithm,
+- services: the list of the services recommended by the system,
+- search_data: search phrase and filters chosen by the user during viewing a services list on the page with the recommendation panel.
+
+User actions and recommendations are key data for learning the machine learning algorithms behind the recommendation system. Because of that, all recommendations and user actions are stored in the recommender database each time they are generated.
+
+Although this crucial data are stored in the recommender DB they have to be properly processed into the form that can be digested by the reinforcement learning-based recommendation algorithms. This form is a SARS.
+
+SARS is a tuple of the following entities: state, action, reward, next state. In the context of the marketplace recommender system each of these entities has the following meaning:
+- state: user, his history of interest in services and his chosen search data (see recommendation),
+- action: services recommended to the user by the recommender system,
+- reward: the value that represents the user's interest in the services recommended to him by the recommender system,
+- next state: user and his history of interest in services and his chosen search data after seeing a recommendation.
+
+The process of formation of SARSes out of the recommendations, user actions and user's ordering history is called SARSes generation and it is carried out by the SARSes generator.
+
+# SARSes generator
+
+As we could see earlier, the user actions can form a tree. Each such tree has a root in the visit on the page with the recommendation panel. Because recommendation visit IDs are compatible with user action visit IDs, if the source visit ID of a user action is the same as the visit ID of a recommendation, it indicates that the user performed this user action after seeing the recommendation. It is a user reaction to the recommendation. It will be used to measure user interest in the recommendation. If he has done more user actions that followed the first one, all of them are important. The whole user actions tree is a reaction to the recommendation. Where does it end? It ends when all branches of the given user action tree end. When the branch of the user action tree ends? It ends at the moment of seeing the next recommendation by the user (then the whole process starts again but for the next recommendation and SARS) or when the user orders a service or when there is no next user action.
```

Comment:
"Because recommendation visit IDs are compatible with user action visit IDs, if the source visit ID of a user action is the same as the visit ID of a recommendation, it indicates that the user performed this user action after seeing the recommendation." This sentence is not clear